### PR TITLE
Minor Cleanup

### PR DESCRIPTION
**Pull request by vivekmore**
Remove common code out of if-else construct

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/aws-containers/prompts.js [hunk 1]
```diff
@@ -281,8 +278,7 @@ function promptPerformance(config, awsConfig = { performance: 'low' }) {
     ];
 
     return this.prompt(prompts).then(props => {
-        const performance = props.performance;
-        return performance;
+        return props.performance;
```
Comment (mshima) [inline generators/aws-containers/prompts.js]: I don't see a reason for 
```
.then(props => {
         return props.performance;
})
```

`return this.prompt(prompts)`
should be enough.

**Comment by mshima** [comment]
Those 
```
const done = this.async();
done();
```
aren't necessary.

returning the the promise (`return this.prompt();`) should be enough.
See https://github.com/jhipster/generator-jhipster/pull/11254

File: generators/aws-containers/prompts.js (hunk 1 above)
Comment (vivekmore) [reply generators/aws-containers/prompts.js]: I was merely removing the declaration of a redundant const.

I can change that to your proposed suggestion, however wouldn't that change the semantics of the existing logic?

Comment (mshima) [reply generators/aws-containers/prompts.js]: This promise isn't used anywhere.
https://github.com/jhipster/generator-jhipster/search?q=promptPerformance&unscoped_q=promptPerformance

Comment (vivekmore) [reply generators/aws-containers/prompts.js]: Ah! Makes sense now. 👍

Comment (vivekmore) [reply generators/aws-containers/prompts.js]: Sorry for asking again but, wouldn't it break https://github.com/jhipster/generator-jhipster/blob/d3cc5f6b8e021422bbab6d8151172ececb0eaf2d/generators/aws-containers/prompts.js#L231
`.then(performance => {`

Comment (mshima) [reply generators/aws-containers/prompts.js]: You are right.

**Comment by vivekmore** [comment]
> Returning the the promise (`return this.prompt();`) should be enough.

I've removed usages of `this.async()`. I hope that's what you meant.

**Comment by mshima** [comment]
Looks great, we just need to execute it to make sure all prompts are waiting for input correctly.

**Comment by vivekmore** [comment]
> Looks great, we just need to execute it to make sure all prompts are waiting for input correctly.

Hopefully the ci tests would validate that.

**Comment by mshima** [comment]
> > Looks great, we just need to execute it to make sure all prompts are waiting for input correctly.
> 
> Hopefully the ci tests would validate that.

It doesn't.
It only tests if the value are been properly set.
I am pretty sure it's correct, but I could have missed something.

**Comment by mshima** [comment]
It needs microservice to test which I don’t use. 🤦‍♂ 
There is no test for aws-constainers.

**Comment by mshima** [comment]
@vivekmore I tried to test, but I failed.
It needs too many config/expertise I don't have.

**Comment by SudharakaP** [comment]
@mshima @vivekmore : Hey, I can help test this one. 😄 

@PierreBesson : Is there a possibility to use that AWS credits/account that we got? Since the AWS containers sub-generator always incurs some cost. If it's not ready yet that's fine too I can create a new account for now. 😄

**Comment by PierreBesson** [comment]
@sudharakaP I will send you an email later with instructions to use our AWS account.

**Comment by SudharakaP** [comment]
@PierreBesson : Wonderful; thanks a bunch 👍

**Comment by SudharakaP** [comment]
@PierreBesson : Just pinging you in case you forgot. :smile:

**Comment by vivekmore** [comment]
@mshima thanks for re-reviewing.
I'm not sure I can merge, since I'm not a part of the jhipster core team.

**Comment by mshima** [comment]
🤦‍♂️

**Comment by vivekmore** [comment]
I relied on the tests so far.

I haven't checked by running the actual generator locally.
I'll check and get back.

**Comment by mshima** [comment]
> I relied on the tests so far.
> 
> I haven't checked by running the actual generator locally.
> I'll check and get back.

Test will not catch errors with the prompt.

**Comment by vivekmore** [comment]
> > I relied on the tests so far.
> > I haven't checked by running the actual generator locally.
> > I'll check and get back.
> 
> Test will not catch errors with the prompt.

I understand and hence said that I'll get back.

What I don't understand is the need to use facepalm 🤦 emoji.

> 🤦‍♂️ @vivekmore the prompts shows correctly?

It wasn't clear from your earlier response, that you wanted me to test:

> @vivekmore feel free to merge this.
> All methods are returning a promise (prompt return).
> This is correct.

I'm also at a loss why you're asking me to do this after so long, when you originally approved without the need for me to test locally. 
If you disagree something with my contribution, please provide feedback without showing frustration towards me.

**Comment by mshima** [comment]
> I understand and hence said that I'll get back.
> 
> What I don't understand is the need to use facepalm 🤦‍♂️ emoji.

Because I didn't see not a core developer.
Sorry about that.

> 
> > 🤦‍♂️ @vivekmore the prompts shows correctly?
> 
> It wasn't clear from your earlier response, that you wanted me to test:
> 
> I'm also at a loss why you're asking me to do this after so long, when you originally approved without the need for me to test locally.
> If you disagree something with my contribution, please provide feedback without showing frustration towards me.

The frustration was about myself.
This patch is correct, I just want want a manual test to guarantee the prompts are waiting for user responses.
I don't have the tools to get to the point the prompts are shown.
I am even considering doing some modifications to yeoman-generator to only run some priorities to get this merged.

Sorry about the misunderstanding.

**Comment by SudharakaP** [comment]
@PierreBesson : Just following up in case we could use the AWS credentials to test this. :smile: 

@mashima, @vivekmore : Hey, let me try to do a manual test today for this and will get back to you tomorrow. I have an idea on how to test this locally which I want to try out. :thinking:

**Comment by PierreBesson** [comment]
@SudharakaP sorry I have been trying to reach out to you to give you access to the JHipster AWS account. Can you send me an email (pierre.besson@jhipster.tech) so that we can arrange a call about it.

**Comment by SudharakaP** [comment]
> @SudharakaP sorry I have been trying to reach out to you to give you access to the JHipster AWS account.

Do you mean my email didn't work or you had trouble sending emails to it? :fearful:

**Comment by vivekmore** [comment]
@mshima I just did a quick sanity check in my local instance using the following steps:

In the branch of PR
```
npm link
```

In a separate empty directory:
```
jhipster
```

All default prompts completed successfully.
Let me know if you'd like me to test further or deeper.

**Comment by mshima** [comment]
@vivekmore that's enough, thanks.
I will just wait till tomorrow if @SudharakaP manages to test too.

**Comment by SudharakaP** [comment]
I have tested this using [Localstack](https://github.com/localstack/localstack) which I discovered recently. :smile: This is not a comprehensive test, but given we are only concerned with the prompts I believe this is adequate. The prompts do wait and I see no other side effect. Therefore I am merging this one. :smile:

**Comment by mshima** [comment]
Thanks @SudharakaP

**Comment by SudharakaP** [comment]
You are welcome. :smile:

**Pull request merged**